connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/CatalystDataToProtobuf.scala

```diff
@@ -26,14 +26,17 @@ import org.apache.spark.sql.types.{BinaryType, DataType}
 private[sql] case class CatalystDataToProtobuf(
     child: Expression,
     messageName: String,
-    descFilePath: Option[String] = None,
+    binaryFileDescriptorSet: Option[Array[Byte]] = None,
     options: Map[String, String] = Map.empty)
     extends UnaryExpression {
 
+  // TODO(SPARK-43578): binaryFileDescriptorSet could be very large in some cases. It is better
```

Comment:
Will `connect` also be changed like this TODO?

Reply:
May be not. It doesn't save much, if any at all (we need to transfer the buffer to the server). 
We can treat those two separately even if we want handle on connect side.